spark/v3.4/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestRewriteDataFilesProcedure.java

```diff
@@ -395,6 +395,37 @@ public void testRewriteDataFilesWithFilterOnPartitionTable() {
     assertEquals("Data after compaction should not change", expectedRecords, actualRecords);
   }
 
+  @Test
+  public void testRewriteDataFilesWithFilterOnOnBucketExpression() {
+    // The schema `system` cannot be found in spark_catalog
+    Assume.assumeFalse(catalogName.equals(SparkCatalogConfig.SPARK.catalogName()));
+    createBucketPartitionTable();
```

Comment:
Same here, maybe we could use the test utils class SystemFunctionPushDownHelper to build the table and data as well.

Reply:
I kept this one to be similar to what we had above, but leveraged your test utils in `testRewriteDataFilesWithPossibleV2Filters` below!